### PR TITLE
adds hashed route to map list: /map#list to enable browser back-button return to the list

### PR DESCRIPTION
**Pull request by thomhickey**
Fixes https://github.com/sfbrigade/support-sfusd/issues/225 by creating a hashed route /map#list that the app uses when in list view mode. On load of the Map component, we check the route and if #list is there we go to the list view. This fixes back-button issue where a user goes to list view, clicks/taps through to a school, then uses browser back button to return to the list. NOTE: the expanded school is not re-expanded in this PR, if we want that functionality we can create another user story.

### CODE REVIEW

**Comment by nickvisut** [comment]
lgtm! Q though: Was the bit at the end with the CSS classes removed intentionally?

**Comment by nickvisut** [comment]
A few of us took a look at this during the eng meeting and tested it a bit. Great job. We could maybe use non-hash paths in the future, but works well enough, so may not be a need.

**Review by nickvisut — APPROVED** [review]
works well as commented above

**Review by mattgianni — COMMENTED** [review]
looks great to me!

**Comment by nickvisut** [comment]
@thomhickey was the bit at the end removed intentionally? (it was originally meant for correcting mobile height going b/t diff pages)

**Comment by thomhickey** [comment]
> @thomhickey was the bit at the end removed intentionally? (it was originally meant for correcting mobile height going b/t diff pages)
> 
> (I meant to ask for this to be addressed before merging, oops)

good catch @nickvisut ... no it was not intentional to remove it. i can put it back if we feel it's needed but leaving the root element as 100dvh seems like what we want for both map and list views as the list is a scrollable list within a fixed height container so the container can be 100dvh. at least that's what i'm seeing, but the code was probably put there to fix a specific case i'm not seeing so we should probably put it back. are you seeing the problem that that old code fixes?

File: src/pages/map.tsx [hunk 1]
```diff
@@ -51,17 +58,14 @@ const Map: React.FC<Props> = (props) => {
   };
 
   const setToggle = () => {
-    setIsMap(!isMap);
+    const newIsMap = !isMap;
```
Comment (nickvisut) [inline src/pages/map.tsx:61]: adding comment for reference link

**Pull request merged**